spark-bigquery-tests/src/main/java/com/google/cloud/spark/bigquery/integration/QueryPushdownIntegrationTestBase.java

```diff
@@ -468,4 +431,597 @@ public void testAggregateExpressions() {
+      assertThat(r.get(0)).isEqualTo(r.get(3));
+    }
+  }
+
+  @Test
+  public void testFullOuterJoin() {
+    writeTestDataToBigQuery(
+        getNumStructDataFrame(TestConstants.numStructDataset),
+        testDataset.toString() + "." + testTable);
+    Dataset<Row> df = readTestDataFromBigQuery(testDataset.toString() + "." + testTable);
+
+    writeTestDataToBigQuery(
+        getNumStructDataFrame(TestConstants.numStructDatasetForJoin),
+        testDataset.toString() + "." + testTable + "_to_join");
+    Dataset<Row> df_to_join =
+        readTestDataFromBigQuery(testDataset.toString() + "." + testTable + "_to_join");
+
+    // disabling pushdown to collect the join result to compare with pushdown enabled
+    BigQueryConnectorUtils.disablePushdownSession(spark);
+    Dataset<Row> dfWithoutPushDownResult =
+        df.alias("num_struct")
+            .join(
+                df_to_join.alias("num_struct_to_join"),
+                df.col("num1").equalTo(df_to_join.col("num1")),
+                "fullouter")
+            .select(
+                "num_struct.num1",
+                "num_struct.num2",
+                "num_struct.num3",
+                "num_struct_to_join.num1",
+                "num_struct_to_join.num2",
+                "num_struct_to_join.num3");
+    List<Row> withoutPushDownResult = dfWithoutPushDownResult.collectAsList();
+
+    // enabling pushdown to test join
+    BigQueryConnectorUtils.enablePushdownSession(spark);
+    Dataset<Row> dfWithPushDownResult =
+        df.alias("num_struct")
+            .join(
+                df_to_join.alias("num_struct_to_join"),
+                df.col("num1").equalTo(df_to_join.col("num1")),
+                "fullouter")
+            .select(
+                "num_struct.num1",
+                "num_struct.num2",
+                "num_struct.num3",
+                "num_struct_to_join.num1",
+                "num_struct_to_join.num2",
+                "num_struct_to_join.num3");
+    List<Row> withPushDownResult = dfWithPushDownResult.collectAsList();
+    // checking if the results with and without pushdown is the same
+    assertThat(withPushDownResult.size()).isEqualTo(withoutPushDownResult.size());
+    assertThat(withoutPushDownResult.containsAll(withPushDownResult)).isTrue();
+    assertThat(withPushDownResult.containsAll(withoutPushDownResult)).isTrue();
+    /*
+      +----+----+----+----+----+----+
+      |num1|num2|num3|num1|num2|num3|
+      +----+----+----+----+----+----+
+      |   3|   2|   1|   3|   4|   1|
+      |   4|   3|   2|   4|   1|   3|
+      |null|null|null|   6|   5|   3|
+      +----+----+----+----+----+----+
+    */
+    assertThat(withPushDownResult.size()).isEqualTo(3);
+    for (Row r : withPushDownResult) {
+      assertThat(r.size()).isEqualTo(6);
+      if (r.get(0) == null) {
+        assertThat(r.get(3)).isEqualTo(6);
+      } else {
+        assertThat(r.get(0)).isEqualTo(r.get(3));
+      }
+    }
+
+    List<Row> result =
```

Comment:
Can you add a comment about swapping tables for join here and in other join tests

Reply:
Added